### ISSUE TITLE
Question about the DPR annotation in the NQ train and eval set

### ISSUE THREAD

**Issue by Xiaoqianhou** [KNOT-FIT-BUT/scalingQA#2]
Hello~ I'd like to use your repo to do some cross validation, that means I need to transform the NQ test set into the same format as the NQ train and eval set. But I found our that the original NQ training set you provided have some DPR annotations like 'multi_span_answers', 'is_mapped', but the original test set don't have these domains. So could you please kindly explain to me how to process the NQ test set so that I can make it the same format like the train and eval set, and is able to fit in the retriever (specifically, run it using extract_DPR_predictions.py)
Thank you very much in advance!

**Comment by MFajcik** [comment]
Hello @Xiaoqianhou,
unfortunately, I do not have a simple answer for you 😞.

The testing set is intended for testing and thus contains no "intermediate" annotation. Note that to obtain a set of retrieved passages from DPR, the test data can still be processed via extract_DPR_predictions.py using the `"test_data_file"` key in the config.

NQ train and validation set were parsed out into our ".jsonl" using DPR training files available [here](https://github.com/facebookresearch/DPR/blob/1ee31c6c5335f05c45138161eb5ce2c03023115d/dpr/data/download_data.py#L37)
and original training set, (where it is possible to distinguish between single-span and multi-span answers). The original training set was initially processed by Lee et al. via [this script](https://github.com/google-research/language/blob/master/language/orqa/preprocessing/convert_to_nq_open.py) from the original [NaturalQuestions dataset](https://ai.google.com/research/NaturalQuestions/download) (its simplified version). 

**To get the single/multi-span annotation**, you need to slightly modify Lee et al.'s script.

However, **to get golden passage annotation / strong negative passage annotation**, you might need to either ask DPR authors for the official script --- which was not released last I've checked their repo --- or resort to your own heuristic.
What I've tried in past and it works pretty well is to:    

1. Take the article name from the original NQ dataset for every Q-A pair and get passages with article names from the passage DB. 
2. Then use some matching heuristic (e.g. F1 overlap) to find the most similar passage to the answer and its context from the original dataset.
